community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/state/storeview/StoreViewNodeStoreScan.java

```diff
@@ -107,10 +107,8 @@ public void process( NodeRecord node ) throws FAILURE
                     updates.added( propertyKeyId, value );
                 }
             }
-            if ( updates.hasUpdates() )
-            {
-                propertyUpdatesVisitor.visit( updates.build() );
-            }
+            // TODO: only visit if has label change or label + prop change?
+            propertyUpdatesVisitor.visit( updates.build() );
```

Comment:
if `updates` has _any_ label/property changes at all? We already know that since the node is from the persistent store there are no _changes_. And determining if the changes are actually relevant or not is what the implementation of the visitor already does.

What we could do is a crude pre-filtering, where we keep a single boolean flag that we set to true in the `if ( propertyKeyIdFilter.test( propertyKeyId ) )` block, is that what you mean?

I can't see that hurting, so why not just do that instead of leaving a vague comment in the code that is mostly just going to cause confusion for future readers of this code.